--- origsrc/libbluray-0.2.2/src/file/dl_posix.c	2011-03-31 07:36:49.000000000 -0500
+++ src/libbluray-0.2.2/src/file/dl_posix.c	2012-05-29 12:47:08.659125500 -0500
@@ -62,7 +62,7 @@ void   *dl_dlopen  ( const char* path, c
 #if defined(__APPLE__)
     static const char ext[] = ".dylib";
     version = NULL;
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
     static const char ext[] = ".dll";
     version = NULL;
 #else
@@ -70,7 +70,18 @@ void   *dl_dlopen  ( const char* path, c
 #endif
 
     if (version) {
+#if defined(__CYGWIN__)
+        char *cygpath = strdup (path);
+        if ((strncmp (path, "lib", 3) == 0)) {
+            cygpath[0] = 'c';
+            cygpath[1] = 'y';
+            cygpath[2] = 'g';
+        }
+        name = str_printf("%s-%s.dll", cygpath, version);
+        free (cygpath);
+#else
         name = str_printf("%s%s.%s", path, ext, version);
+#endif
     } else {
         name = str_printf("%s%s", path, ext);
     }
--- origsrc/libbluray-0.2.2/src/libbluray/bdj/bdj.c	2011-03-31 07:36:49.000000000 -0500
+++ src/libbluray-0.2.2/src/libbluray/bdj/bdj.c	2012-05-29 12:38:54.489860600 -0500
@@ -172,6 +172,8 @@ void* load_jvm()
 
 #ifdef WIN32
     char* path = str_printf("%s/jre/bin/server/jvm", java_home);
+#elif defined(__CYGWIN__)
+    char* path = str_printf("%s/bin/cygjvm", java_home);
 #else	//	#ifdef WIN32
     char* path = str_printf("%s/jre/lib/%s/server/libjvm", java_home, JAVA_ARCH);
 #endif	//	#ifdef WIN32
--- origsrc/libbluray-0.2.2/src/Makefile.am	2012-03-15 11:02:31.000000000 -0500
+++ src/libbluray-0.2.2/src/Makefile.am	2012-05-29 13:13:29.006516200 -0500
@@ -88,7 +88,7 @@ libbluray_la_SOURCES=libbluray/bluray.h
 	util/bits.h \
 	util/logging.h
 libbluray_la_HEADERS= libbluray/bluray.h file/filesystem.h util/log_control.h libbluray/keys.h libbluray/decoders/overlay.h libbluray/bdnav/meta_data.h libbluray/bdnav/clpi_data.h libbluray/bluray-version.h
-libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) $(DLOPEN_LDFLAGS)
+libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) -no-undefined $(DLOPEN_LDFLAGS)
 libbluray_la_LIBADD= $(LIBXML2_LIBS)
 
 if USING_BDJAVA
